### PR TITLE
Servicefabrictemplates

### PR DESCRIPTION
**Pull request by ChackDan**
5 Node Service Fabric Cluster template

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@ChackDan__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (chackdan). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, or work for Microsoft Open Technologies, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

File: service-fabric-cluster-5-node-1-nodetype/azuredeploy.json [hunk 1]
```diff
@@ -0,0 +1,592 @@
...
+    "adminUserName": {
+      "type": "string",
+      "metadata": {
+        "description": "Remote Desktop user ID"
+      }
+    },
+    "adminPassword": {
+      "type": "securestring",
+      "metadata": {
+        "description": "Remote Desktop user Password- must be 8 + characters long, must contain alpabets, numbers, ok to use special chars"
```
Comment (ChackDan) [inline service-fabric-cluster-5-node-1-nodetype/azuredeploy.json]: This descriptions is too long. I will fix this.

File: service-fabric-cluster-5-node-1-nodetype/azuredeploy.json [hunk 2]
```diff
@@ -0,0 +1,592 @@
+{
+  "$schema": "http://schema.management.azure.com/schemas/2014-04-01-preview/deploymentTemplate.json",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "clusterLocation": {
+      "type": "string",
+      "allowedValues": [
```
Comment (singhkay) [inline service-fabric-cluster-5-node-1-nodetype/azuredeploy.json]: This should change to more readable values of type "West US"

File: service-fabric-cluster-5-node-1-nodetype/azuredeploy.json [hunk 3]
```diff
@@ -0,0 +1,592 @@
...
+        "eastus"
+      ],
+      "metadata": {
+        "description": "Location of the Cluster"
+      }
+    },
+    "clusterName": {
+      "type": "string",
+      "metadata": {
+        "description": "Name of your cluster - Between 3 and 23 characters.  Lowercase letters and numbers"
```
Comment (singhkay) [inline service-fabric-cluster-5-node-1-nodetype/azuredeploy.json]: Will the deployment fail if a user enters a capital letter? If so, then I suggest creating a variable that converts the value in the parameter here to all lower case using the toLower function

File: service-fabric-cluster-5-node-1-nodetype/azuredeploy.json [hunk 4]
```diff
@@ -0,0 +1,592 @@
...
+      "lbIPConfig0": "[concat(variables('lbID0'),'/frontendIPConfigurations/LoadBalancerIPConfig')]",
+      "lbPoolID0": "[concat(variables('lbID0'),'/backendAddressPools/LoadBalancerBEAddressPool')]",
+      "lbProbeID0": "[concat(variables('lbID0'),'/probes/FabricGatewayProbe')]",
+      "lbHttpProbeID0": "[concat(variables('lbID0'),'/probes/FabricHttpGatewayProbe')]",
+      "vmNodeType0Name": "[concat('NT1', variables('vmName'))]",
+      "vmNodeType0Size": "Standard_A2"
+    },
+    "resources": [
+      {
+        "apiVersion": "2015-05-01-preview",
```
Comment (singhkay) [inline service-fabric-cluster-5-node-1-nodetype/azuredeploy.json]: Consider making the api version a variable so it's easy to move to newer api versions across all resources below. Also try using the newer 2015-06-15 API

**Pull request merged**